File: Android/tuilivekit/src/main/java/com/trtc/uikit/livekit/state/view/ViewState.java
package com.trtc.uikit.livekit.state.view;

import static com.trtc.uikit.livekit.state.LiveDefine.LinkStatus.NONE;

import com.trtc.tuikit.common.livedata.LiveData;
import com.trtc.uikit.livekit.state.LiveDefine;

public class ViewState {
    public LiveData<LiveDefine.LinkStatus> linkStatus             = new LiveData<>(NONE);
    public LiveData<LiveDefine.LiveStatus> liveStatus             = new LiveData<>(LiveDefine.LiveStatus.NONE);
    public LiveData<Boolean>               autoOpenCameraOnSeated = new LiveData<>(true);

    public void reset() {
        linkStatus.set(NONE);
        autoOpenCameraOnSeated.set(true);
        liveStatus.set(LiveDefine.LiveStatus.NONE);
    }
}

File: Flutter/lib/state/view/view_state.dart
import 'package:flutter/material.dart';
import 'package:tencent_live_uikit/state/live_define.dart';

class ViewState {
  final ValueNotifier<LinkStatus> linkStatus = ValueNotifier<LinkStatus>(LinkStatus.none);
  final ValueNotifier<bool> autoOpenCameraOnSeated = ValueNotifier<bool>(false);
  final ValueNotifier<LiveStatus> liveStatus = ValueNotifier<LiveStatus>(LiveStatus.none);

  void reset() {
    linkStatus.value = LinkStatus.none;
    autoOpenCameraOnSeated.value = false;
    liveStatus.value = LiveStatus.none;
  }
}
